api/src/Repository/Model/BaseModel.php

```diff
@@ -206,10 +207,10 @@ public function push()
 
             if ($this->isCollection($models)) {
                 /** @var Collection $models */
-                $models = $models->all(true);
+                $modelsArray = $models->all(true);
                 $error = false;
                 $errors = [];
-                foreach (array_filter($models) as $model) {
+                foreach (array_filter($modelsArray) as $model) {
```

Comment:
We need to revisit this entire file - we might not need any of it's functions as we are not doing model-level validation.